cpp/models/ide_secir/model.cpp

```diff
@@ -29,64 +29,90 @@ namespace mio
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSevere)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedCritical)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)];
+        }
+        else if (m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Susceptible)] > 1e-12) {
+            //take initialized value for Susceptibles if value can't be calculated via the standard formula
+            //calculate other compartment sizes for t=0
+            other_compartments_current_timestep(dt);
+
+            //R; need an initial value for R, therefore do not calculate via compute_recovered()
+            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Recovered)] =
+                m_N - m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Susceptible)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Exposed)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedNoSymptoms)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSymptoms)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSevere)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedCritical)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)];
+        }
+        else if (m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Recovered)] > 1e-12) {
+            //if value for Recovered is initialized and standard method is not applicable, calculate Susceptibles via other compartments
+            //determining other compartment sizes is not dependent of Susceptibles(0), just of the transitions of the past.
+            //calculate other compartment sizes for t=0
+            other_compartments_current_timestep(dt);
+
+            m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Susceptible)] =
+                m_N - m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Exposed)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedNoSymptoms)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSymptoms)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedSevere)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::InfectedCritical)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Recovered)] -
+                m_populations[Eigen::Index(0)][Eigen::Index(InfectionState::Dead)];
+        }
+        else {
+            log_error("Error occured while initializing compartments: Force of infection is evaluated to 0 and neither "
+                      "Susceptibles nor Recovered for time 0 were set. One of them should be larger 0.");
+        }
```

Comment:
I think the error message should also include total_confirmed_cases.
Also, I would find it more intuitive if the distinction between `m_total_confirmed_cases < 1e-12` and the other case was the other way around, i.e. if it would be checked if `m_total_confirmed_cases > 1e-12` first.